[8.13][Security Solution][Endpoint] Add beta badge to sentinel one connector cards/flyout and responder/isolation action flyouts

ashokaditya opened the pull request:
## Summary

Replaces Technical Preview badge with a Beta badge for Sentinel One connector cards and connector form flyouts.
Additionally it shows Beta badge on Responder for sentinel one alerts as well as Isolate/Release action flyouts.

### TODO/DONE
- [x] Update beta badge tooltip text on all used instances
- [x] beta badges on alerts flyout/responder behind feature flag

### Screens

### Connectors list flyout
<img width="2054" alt="Screenshot 2024-02-05 at 2 14 12 PM" src="https://github.com/elastic/kibana/assets/1849116/8aa3215a-ebc0-4935-92fe-5d234f78e17b">

### Sentinel one connector forms
#### Create
<img width="2056" alt="Screenshot 2024-02-05 at 2 14 26 PM" src="https://github.com/elastic/kibana/assets/1849116/fae12ae7-5301-4bc0-8934-3010b823ec5b">

#### Edit
<img width="2054" alt="Screenshot 2024-02-05 at 2 31 04 PM" src="https://github.com/elastic/kibana/assets/1849116/637f1d01-fe59-47b4-bb8e-714dd9304c58">

#### Isolate/Release flyout forms for SentinelOne
![Screenshot 2024-02-07 at 1 03 32 PM](https://github.com/elastic/kibana/assets/1849116/92f238e7-4d9c-45aa-8b73-40fdc24dea43)
![Screenshot 2024-02-07 at 1 03 13 PM](https://github.com/elastic/kibana/assets/1849116/ec0b5afc-88f9-424b-992c-5d65076d2490)


#### Response Console for SentinelOne
![Screenshot 2024-02-07 at 11 51 22 AM](https://github.com/elastic/kibana/assets/1849116/4b5b9a52-5493-4ae2-8f77-65bed1a7d182)

### Checklist

- [x] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/main/packages/kbn-i18n/README.md)
- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios
- [ ] [Flaky Test Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was used on any tests changed
- [x] Any UI touched in this PR does not create any new axe failures (run axe in browser: [FF](https://addons.mozilla.org/en-US/firefox/addon/axe-devtools/), [Chrome](https://chrome.google.com/webstore/detail/axe-web-accessibility-tes/lhdoppojpmngadmnindnejefpokejbdd?hl=en-US))
- [x] This renders correctly on smaller devices using a responsive layout. (You can test this [in your browser](https://www.browserstack.com/guide/responsive-testing-on-local-server))
- [x] This was checked for [cross-browser compatibility](https://www.elastic.co/support/matrix#matrix_browsers)

ashokaditya committed:
add beta badge to sentinel one connector cards/flyout

closes elastic/security-team/issues/8526

ashokaditya commented:
/ci

ashokaditya committed:
add beta tag to sentinel one responder

closes elastic/security-team/issues/8526

ashokaditya committed:
Add beta badge to alert isolate/release action flyout

closes elastic/security-team/issues/8526

ashokaditya committed:
Update header.test.tsx

refs 0bfd258167ba3d8b110b1ff6c22dcdbbdb4a0272

ashokaditya commented on x-pack/plugins/security_solution/public/common/translations.ts:
```diff
@@ -19,6 +19,11 @@ export const BETA = i18n.translate('xpack.securitySolution.pages.common.beta', {
 
+export const BETA_TOOLTIP = i18n.translate('xpack.securitySolution.pages.common.beta.tooltip', {
+  defaultMessage:
+    'This functionality is in beta and may be changed or removed completely in a future release. Elastic will work to fix any issues, but features in beta are not subject to the support SLA of official GA features.',
```
@caitlinbetz @dasansol92 @paul-tavares This needs an update.

ashokaditya replied:
Updated it based on https://www.elastic.co/guide/en/kibana/8.12/infrastructure.html#infrastructure

elasticmachine commented:
Pinging @elastic/security-defend-workflows (Team:Defend Workflows)

ashokaditya committed:
Update beta badge tooltip text

ashokaditya commented on x-pack/plugins/triggers_actions_ui/public/application/sections/action_connector_form/beta_badge_props.tsx:
```diff
@@ -8,6 +8,16 @@
+  }),
+  tooltipContent: i18n.translate('xpack.triggersActionsUI.betaBadgeDescription', {
+    defaultMessage:
+      'This functionality is in beta and may be changed or removed completely in a future release. Elastic will work to fix any issues, but features in beta are not subject to the support SLA of official GA features.',
```
Updated it based on https://www.elastic.co/guide/en/kibana/8.12/infrastructure.html#infrastructure.

logeekal commented on x-pack/plugins/security_solution/public/flyout/document_details/isolate_host/header.test.tsx:
```diff
@@ -39,4 +43,19 @@ describe('<PanelHeader />', () => {
     expect(getByTestId(FLYOUT_HEADER_TITLE_TEST_ID)).toHaveTextContent('Release host');
   });
+
+  it.each(['isolateHost', 'unisolateHost'])(
```
Does it make sense to add a small test to make sure of the non-existence of `beta` badge is it is not a Sentinal one alert.

ashokaditya replied:
Absolutely

logeekal replied:
Looks good. Thank you. Resolving this comment.

logeekal reviewed (approved):
Investigation changes LGTM. Thank you. Added a `nit` comment.

ashokaditya committed:
add tests

review changes @logeekal

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/eslint --no-cache --fix'

ashokaditya committed:
beta badges for responder and isolate/release flyout

enable those with feature flag

ashokaditya committed:
update tests

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/lint_ts_projects --fix'

ashokaditya committed:
fix lint and import

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/lint_ts_projects --fix'

paul-tavares reviewed (approved):
LGTM 🚢 it

ashokaditya committed:
Connector shows tech preview badge without ff

Ensure that tech preview badge shows on sentinel one connector when the new feature flag is not enabled.

So in addition to setting `responseActionsSentinelOneV1Enabled` true under `xpack.securitySolution.enableExperimental` that shows beta badges for responder/isolate/release flyout panels on alerts;

We also need to set `sentinelOneConnectorOnBeta` to true under `xpack.stack_connectors.enableExperimental` to ensure that the connector shows beta badge instead of tech preview

logeekal commented on x-pack/plugins/security_solution/public/flyout/document_details/isolate_host/header.test.tsx:
```diff
@@ -39,4 +49,36 @@ describe('<PanelHeader />', () => {
+    }
+  );
+
+  it.each(['isolateHost', 'unisolateHost'])(
```
Requesting you to add one more test to test impact of `experimental_flag = false` on the visibility of the `beta` badge.

ashokaditya committed:
update tests

review changes @logeekal

logeekal reviewed (approved):
LGTM 🚀 . Thanks for accommodating the changes.

ashokaditya committed:
Merge branch 'main' into task/dw-beta-badge-sentinel-one-8526

ashokaditya committed:
Merge branch 'main' into task/dw-beta-badge-sentinel-one-8526

ashokaditya committed:
Merge branch 'main' into task/dw-beta-badge-sentinel-one-8526

ashokaditya committed:
Merge branch 'main' into task/dw-beta-badge-sentinel-one-8526

ashokaditya committed:
Merge branch 'main' into task/dw-beta-badge-sentinel-one-8526

ashokaditya committed:
Merge branch 'main' into task/dw-beta-badge-sentinel-one-8526

ersin-erdal reviewed (approved):
ResponseOps changes LGTM

ashokaditya committed:
Merge branch 'main' into task/dw-beta-badge-sentinel-one-8526

ashokaditya committed:
Merge branch 'main' into task/dw-beta-badge-sentinel-one-8526
